lib/src/snapd_client.dart

```diff
@@ -488,6 +500,85 @@ class SnapdNotice with _$SnapdNotice {
       _$SnapdNoticeFromJson(json);
 }
 
+/// A class to model a generate-recovery-key response from snapd
+@freezed
+class SnapdGenerateRecoveryKeyResponse with _$SnapdGenerateRecoveryKeyResponse {
+  const factory SnapdGenerateRecoveryKeyResponse({
+    required String recoveryKey,
+    required String opaqueId,
```

Comment:
Looking at the spec and testing against snapd from edge, this should be
```suggestion
    required String keyId,
```